reana_client/cli/workflow.py

```diff
@@ -477,9 +477,71 @@ def workflow_validate(ctx, file):
             err=True)
 
 
+@click.command(
+    'delete',
+    help='Delete a workflow run.')
+@click.option(
+    '-A',
+    '--all',
+    'all_runs',
+    count=True,
+    help='Delete all runs of a given workflow.')
+@click.option(
+    '--hard',
+    'hard_delete',
+    count=True,
+    help='Completely remove workflow run data and workspace from REANA.')
+@click.option(
+    '-w',
+    '--workflow',
+    default=os.environ.get('REANA_WORKON', None),
+    callback=workflow_uuid_or_name,
+    help='Name and run number to be deleted. '
+         'Overrides value of REANA_WORKON.')
+@add_access_token_options
+@click.pass_context
+@with_api_client
+def workflow_delete_run(ctx, workflow, all_runs, hard_delete, access_token):
+    """Delete a workflow run given the workflow name and run number."""
+    logging.debug('command: {}'.format(ctx.command_path.replace(" ", ".")))
+    for p in ctx.params:
+        logging.debug('{param}: {value}'.format(param=p, value=ctx.params[p]))
+
+    if not access_token:
+        click.echo(
+            click.style(ERROR_MESSAGES['missing_access_token'],
+                        fg='red'), err=True)
+        sys.exit(1)
+
+    if workflow:
+        try:
+            logging.info('Connecting to {0}'.format(ctx.obj.client.server_url))
+            response = ctx.obj.client.delete_workflow(workflow,
+                                                      all_runs,
+                                                      hard_delete,
+                                                      access_token)
+            click.echo(
+                click.style('{} has been started.'.format(workflow),
+                            fg='green'))
+
+        except Exception as e:
+            logging.debug(traceback.format_exc())
+            logging.debug(str(e))
+            click.echo(
+                click.style('Workflow could not be started: \n{}'
+                            .format(str(e)), fg='red'),
+                err=True)
+    else:
+        click.secho('Workflow name and number must be provided'
```

Comment:
In case $REANA_WORKON var is not set and --workflow flag is not given, error message is printed twice. Do we want that?

`callback=workflow_uuid_or_name`

```
(reana-client3) rokas:reana-client rokas$ reana-client delete
Workflow name must be provided either with `--workflow` option or with REANA_WORKON environment variable
Workflow name and number must be provided either with `--workflow` option or with REANA_WORKON environment variable
```

Reply:
No, this is happening for all other cli commands in `workflow.py` and it should be fixed with the next commit. This is still wip, but as it seems wafflebot puts it `in review` after every push